### ISSUE TITLE
Rule for preceding a heading with a horizontal rule

### ISSUE THREAD

**Issue by rutrum** [rvben/rumdl#832]
### Problem or motivation

I frequently see LLMs generate horizontal rules right before sections.

```markdown
## Topic

Prose.

---

## Next Topic
```

I think this is redundant, since the next header already marks a transition between the text. 

### Proposed solution

I would like a rule to flag this, and perhaps automatically remove them in formatting.

### Alternatives considered

I could write up a rule in `ast-grep` or something, but it'd be nice if the logic came from a tool that actually parses markdown with AST.

### Feature category

New lint rule

### Example markdown

```markdown
Prose

---

## Next topic

--after formatting/linting-->

Prose

## Next topic
```

### Contribution

- [ ] I would be willing to submit a PR to implement this feature
